src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -582,74 +637,74 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
-        picker.ViewMode(PickerViewMode::Thumbnail);
-        picker.SuggestedStartLocation(PickerLocationId::PicturesLibrary);
-
-        // Converted into a BitmapIconSource. This list of supported image file formats is from BitmapImage documentation
-        // https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Media.Imaging.BitmapImage?view=winrt-19041#remarks
-        picker.FileTypeFilter().ReplaceAll({ L".jpg", L".jpeg", L".png", L".bmp", L".gif", L".tiff", L".ico" });
-
-        StorageFile file = co_await picker.PickSingleFileAsync();
-        if (file != nullptr)
+        auto file = co_await OpenImagePicker();
+        if (!file.empty())
         {
-            _State.Profile().Icon(file.Path());
+            _State.Profile().Icon(file);
         }
     }
 
     fire_and_forget Profiles::Commandline_Click(IInspectable const&, RoutedEventArgs const&)
     {
         auto lifetime = get_strong();
 
-        FileOpenPicker picker;
+        static constexpr COMDLG_FILTERSPEC supportedFileTypes[] = {
+            { L"Executable FIles (*.exe, *.cmd, *.bat)", L"*.exe;*.cmd;*.bat" },
+            { L"All Files (*.*)", L"*.*" }
+        };
 
-        _State.WindowRoot().TryPropagateHostingWindow(picker); // if we don't do this, there's no HWND for it to attach to
-        picker.ViewMode(PickerViewMode::Thumbnail);
-        picker.SuggestedStartLocation(PickerLocationId::ComputerFolder);
-        picker.FileTypeFilter().ReplaceAll({ L".bat", L".exe", L".cmd" });
+        static constexpr winrt::guid clientGuidExecutables{ 0x2E7E4331, 0x0800, 0x48E6, { 0xB0, 0x17, 0xA1, 0x4C, 0xD8, 0x73, 0xDD, 0x58 } };
+        auto path = co_await OpenFilePicker([](auto&& dialog) {
+            THROW_IF_FAILED(dialog->SetClientGuid(clientGuidExecutables));
+            try
+            {
+                auto folderShellItem{ winrt::capture<IShellItem>(&SHGetKnownFolderItem, FOLDERID_ComputerFolder, KF_FLAG_DEFAULT, nullptr) };
+                dialog->SetDefaultFolder(folderShellItem.get());
+            }
+            CATCH_LOG(); // non-fatal
+            THROW_IF_FAILED(dialog->SetFileTypes(ARRAYSIZE(supportedFileTypes), supportedFileTypes));
+            THROW_IF_FAILED(dialog->SetFileTypeIndex(1)); // the array is 1-indexed
+            THROW_IF_FAILED(dialog->SetDefaultExtension(L"exe;cmd;bat"));
+        });
 
-        StorageFile file = co_await picker.PickSingleFileAsync();
-        if (file != nullptr)
+        if (!path.empty())
         {
-            _State.Profile().Commandline(file.Path());
+            _State.Profile().Commandline(path);
         }
     }
 
     fire_and_forget Profiles::StartingDirectory_Click(IInspectable const&, RoutedEventArgs const&)
     {
         auto lifetime = get_strong();
-        FolderPicker picker;
-        _State.WindowRoot().TryPropagateHostingWindow(picker); // if we don't do this, there's no HWND for it to attach to
```

Comment:
Can we revert `TryPropagateHostingWindow`? PR #8391

Reply:
Happy to revert it for 1.9, but not for 1.8 or 1.7.